dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/HibernateTrackedEntityStore.java

```diff
@@ -386,17 +386,18 @@ private String getFromSubQuery(
 
             // INNER JOIN on constraints
             .append(joinPrograms(params))
-            .append(joinAttributeValue(params))
             .append(getFromSubQueryJoinProgramOwnerConditions(params))
             .append(getFromSubQueryJoinOrgUnitConditions(params))
             .append(getFromSubQueryJoinEnrollmentConditions(params))
 
             // LEFT JOIN attributes we need to sort on.
             .append(getFromSubQueryJoinOrderByAttributes(params))
+            .append(getLeftJoinFromFilterConditions(params))
 
             // WHERE
             .append(getFromSubQueryTrackedEntityConditions(whereAnd, params))
-            .append(getFromSubQueryEnrollmentConditions(whereAnd, params));
+            .append(getFromSubQueryEnrollmentConditions(whereAnd, params))
+            .append(getWhereClauseFromFilterConditions(whereAnd, params));
```

Comment:
If it is possible, it would make sense `getWhereClauseFromFilterConditions` to be first in the `where` section to be as close as possible to the related left join